File: app/src/main/java/com/worldventures/dreamtrips/wallet/domain/entity/SmartCardFirmware.java
package com.worldventures.dreamtrips.wallet.domain.entity;

import org.immutables.value.Value;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
public abstract class SmartCardFirmware {

   @Nullable
   public abstract String firmwareBundleVersion();

   @Value.Default
   public String nordicAppVersion() {
      return "";
   }

   @Value.Default
   public String nrfBootloaderVersion() {
      return "";
   }

   @Value.Default
   public String internalAtmelVersion() {
      return "";
   }

   @Value.Default
   public String internalAtmelBootloaderVersion() {
      return "";
   }

   @Value.Default
   public String externalAtmelVersion() {
      return "";
   }

   @Value.Default
   public String externalAtmelBootloaderVersion() {
      return "";
   }

}

File: wallet/src/main/java/com/worldventures/wallet/domain/entity/SmartCardFirmware.kt
package com.worldventures.wallet.domain.entity

import com.esotericsoftware.kryo.DefaultSerializer
import com.esotericsoftware.kryo.serializers.CompatibleFieldSerializer

@DefaultSerializer(CompatibleFieldSerializer::class)
data class SmartCardFirmware(
      val firmwareBundleVersion: String? = null,
      val nordicAppVersion: String = "",
      val nrfBootloaderVersion: String = "",
      val internalAtmelVersion: String = "",
      val internalAtmelBootloaderVersion: String = "",
      val externalAtmelVersion: String = "",
      val externalAtmelBootloaderVersion: String = "") {

   val isEmpty: Boolean
      get() = (nordicAppVersion.isEmpty()
            && nrfBootloaderVersion.isEmpty()
            && internalAtmelVersion.isEmpty()
            && externalAtmelVersion.isEmpty())
}
